Model/Category/Builder.php

```diff
@@ -36,79 +36,54 @@
-use Magento\Catalog\Model\Product;
 use Magento\Framework\Event\ManagerInterface;
-use Magento\Framework\Exception\NoSuchEntityException;
 use Nosto\Tagging\Logger\Logger as NostoLogger;
+use Nosto\Object\Category as NostoCategory;
+use Nosto\Tagging\Model\CategoryString\Builder as NostoCategoryString;
 
 class Builder
 {
     private $logger;
-    private $categoryRepository;
     private $eventManager;
+    private $nostoCategoryString;
 
     /**
-     * @param CategoryRepositoryInterface $categoryRepository
+     * Builder constructor.
      * @param NostoLogger $logger
      * @param ManagerInterface $eventManager
+     * @param NostoCategoryString $nostoCategoryString
      */
     public function __construct(
-        CategoryRepositoryInterface $categoryRepository,
         NostoLogger $logger,
-        ManagerInterface $eventManager
+        ManagerInterface $eventManager,
+        NostoCategoryString $nostoCategoryString
     ) {
-        $this->categoryRepository = $categoryRepository;
+        $this->nostoCategoryString = $nostoCategoryString;
         $this->logger = $logger;
         $this->eventManager = $eventManager;
     }
 
-    /**
-     * @param Product $product
-     * @param Store $store
-     * @return array
-     */
-    public function buildCategories(Product $product, Store $store)
-    {
-        $categories = [];
-        foreach ($product->getCategoryCollection() as $category) {
-            $categoryString = $this->build($category, $store);
-            if (!empty($categoryString)) {
-                $categories[] = $categoryString;
-            }
-        }
-
-        return $categories;
-    }
-
     /**
      * @param Category $category
      * @param Store $store
      * @return null|string
      */
     public function build(Category $category, Store $store)
     {
-        $nostoCategory = '';
+        $nostoCategory = new NostoCategory();
         try {
-            $data = [];
-            $path = $category->getPath();
-            $storeId = $store->getId();
-            foreach (explode('/', $path) as $categoryId) {
-                try {
-                    $category = $this->categoryRepository->get($categoryId, $storeId);
-                } catch (NoSuchEntityException $noSuchEntityException) {
-                    continue;
-                }
-                if ($category instanceof Category
-                    && $category->getLevel() > 1
-                    && !empty($category->getName())
-                ) {
-                    $data[] = $category->getName();
-                }
-            }
-            $nostoCategory = count($data) ? '/' . implode('/', $data) : '';
+            $nostoCategory->setId($category->getId());
+            $nostoCategory->setParentId($category->getParentId());
+            $nostoCategory->setImageUrl($category->getImageUrl());
+            $nostoCategory->setLevel($category->getLevel());
+            $nostoCategory->setUrl($category->getUrl());
```

Comment:
Does the URL reflect the store correctly?

Reply:
Yes it does